src/pages/ErrorPage/NotFoundPage.tsx

```diff
@@ -2,18 +2,38 @@ import React from 'react';
 import type {FullPageNotFoundViewProps} from '@components/BlockingViews/FullPageNotFoundView';
 import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
 import ScreenWrapper from '@components/ScreenWrapper';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import Navigation from '@libs/Navigation/Navigation';
+import * as ReportUtils from '@libs/ReportUtils';
+import ROUTES from '@src/ROUTES';
 
 type NotFoundPageProps = {
     onBackButtonPress?: () => void;
+    isReportRelatedPage?: boolean;
 } & FullPageNotFoundViewProps;
 
 // eslint-disable-next-line rulesdir/no-negated-variables
-function NotFoundPage({onBackButtonPress, ...fullPageNotFoundViewProps}: NotFoundPageProps) {
+function NotFoundPage({onBackButtonPress = () => Navigation.goBack(), isReportRelatedPage, ...fullPageNotFoundViewProps}: NotFoundPageProps) {
+    const {isSmallScreen} = useWindowDimensions();
+
     return (
         <ScreenWrapper testID={NotFoundPage.displayName}>
             <FullPageNotFoundView
                 shouldShow
-                onBackButtonPress={onBackButtonPress}
+                onBackButtonPress={() => {
+                    if (!isReportRelatedPage || !isSmallScreen) {
+                        onBackButtonPress();
+                        return;
+                    }
+                    const topmostReportId = Navigation.getTopmostReportId();
+                    const report = ReportUtils.getReport(topmostReportId ?? '');
+                    // detect the report is invalid
+                    if (topmostReportId && (!report || report.errorFields?.notFound)) {
+                        Navigation.goBack(ROUTES.HOME, true, true);
```

Comment:
@daledah Thanks for the PR.

In the Not Found page, we also have "Go back to homepage" text link, if I am not wrong, on pressing it, it will take us to the home page dismissing the modal. Can we use the same logic for the back button as well?